src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/hooks/useOnChange.ts

```diff
@@ -18,30 +19,47 @@ type Props = {
   data: TableData
   sectionName: string
 }
+export type OnChangeTaxon = (value: Taxon | string) => void
+export type OnChangeDefault = (
+  event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>
+) => void
+type OnChange = OnChangeTaxon | OnChangeDefault
 
 type UseOnChange = {
-  onChange: ChangeEventHandler<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>
-  onPaste: ClipboardEventHandler<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>
+  onChange: OnChange
+  onPaste: React.ClipboardEventHandler<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>
 }
 
 export default (props: Props): UseOnChange => {
   const { table, col, row, nodeValue, data, sectionName } = props
+  const type = col.props.colType
+
   const dispatch = useAppDispatch()
   const countryIso = useCountryIso()
   const cycle = useCycle()
   const assessment = useAssessment()
   const assessmentSection = useAssessmentSection(sectionName)
 
-  const _persistSanitizedValue = (value: string) => {
-    const type = col.props.colType
-    if (Sanitizer.isAcceptable({ type, value })) {
+  const _persistSanitizedValue = (value: string | Taxon) => {
+    const _value = typeof value === 'string' ? value : value?.scientificName
```

Comment:
this is generic to taxon. it doesn't fit here

Reply:
This will be removed in refactor so we can pass the nodeValue directly; we have to modify the nodeValue in the onChange, that's why it's here